Read only secondaries (#396)
neo4j/docs-aura: modules/ROOT/pages/auradb/managing-databases/read-only-secondaries.adoc

[[aura-read-only-secondaries]]
= Read-only secondary instances

> :description: The main responsibility of Read-only secondaries is to scale out read workloads.

Reviewer: ```suggestion
:description: This section describes the use of secondary instances for scaling out read workloads.
```

Read-only secondaries are distributed across availability zones to ensure high availability, performance, and cost efficiency. 
For workloads with heavy read operations, these secondaries prevent bottlenecks and allow batch processing.

Up to 15 static read-only secondaries can be added per database instance, significantly increasing read capacity to handle read-heavy workloads. 
These secondaries can be added, managed, and removed through the Aura Console or the Aura API.

== Read-only secondaries info:

	•	*Availability zones:* Distributed across availability zones to maintain high availability and performance.
	•	*Static configuration:* Currently, read-only secondaries are static and do not support elastic or auto-scaling behavior.
	•	*Region constraints:* Available only within the same region as their primary database instance (single region).
	•	*Pause/Resume:* When an AuraDB Enterprise database is paused and resumed, the count of read-only secondaries is retained. For example, if a database with 3 secondaries is paused, it will resume with the same number of secondaries.

[CAUTION]
====

> *Note:* Secondaries are currently not included in consumption reports.

Reviewer: ```suggestion
Secondaries are currently not included in consumption reports.
```

It allows to further balance the READ and improve scalability.

== Aura Provisioning API - simple request 

Using the Aura API, secondaries can be launched, managed, or terminated.

Once enabled for the user’s tenant a GET instance details call will show secondary count.

[source]
----

GET/instances/{instanceId]{
  "data": {
    "cloud_provider": "azure",
    "connection_url": "neo4j+s://xxxxxx.databases.neo4j.io",
    "id": "xxxxxx",
    "memory": "2GB",
    "name": "Neo4j",
    "region": "uk south",
    "secondaries_count": 0,
    "status": "running",
    "storage": "4GB",
    "tenant_id": "XXXXXXXXXXXXXXXXXXXX",
    "type": "enterprise-db"
  }
}

----

PATCH/instances/{instanceId]
{
  "secondaries_count": 2
}'

[source]
----

A simple Secondaries count patch request will request update {
  "data": {
    "cloud_provider": "azure",
    "connection_url": "neo4j+s://xxxxxx.databases.neo4j.io",
    "id": "xxxxxx",
    "memory": "2GB",
    "name": "Neo4j",
    "region": "uk south",
    "secondaries_count": 2,
    "status": "running",
    "storage": "4GB",
    "tenant_id": "XXXXXXXXXXXXXXXXXXXXX",
    "type": "enterprise-db"
  }
}

> 

Reviewer: ```suggestion
----
```
